apm-agent-plugins/apm-hibernate-search-plugin/apm-hibernate-search-plugin-5_x/src/main/java/co/elastic/apm/agent/hibernate/search/v5_x/HibernateSearch5Instrumentation.java

```diff
@@ -0,0 +1,82 @@
+package co.elastic.apm.agent.hibernate.search.v5_x;
+
+import java.util.Collection;
+import java.util.Collections;
+
+import javax.annotation.Nullable;
+
+import co.elastic.apm.agent.bci.ElasticApmInstrumentation;
+import co.elastic.apm.agent.bci.VisibleForAdvice;
+import co.elastic.apm.agent.hibernate.search.HibernateSearchConstants;
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.hibernate.search.query.hibernate.impl.FullTextQueryImpl;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+public class HibernateSearch5Instrumentation extends ElasticApmInstrumentation {
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return HibernateSearch5ExecuteAdvice.class;
+    }
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return named("org.hibernate.search.query.hibernate.impl.FullTextQueryImpl");
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return named("list");
```

Comment:
What about other `org.hibernate.Query` relevant functions like `scroll` and `iterate`?

Reply:
We don't use those methods in our application because they are not available when you have a JPA FullTextQuery. But none the less a valid addition. I only wonder if the time can be correctly tracked because both methods only load the result when requested. Was this already done in any other plugin already.

Reply:
> I only wonder if the time can be correctly tracked because both methods only load the result when requested. Was this already done in any other plugin already.

Well, I believe this is the case with JDBC's ResultSet, which we do not instrument either (but only the query execution). We miss info about actual reads this way, but the alternative is having a LOT of spans traced, which are largely non-informative.

I assume these methods' duration account for the actual search action (without the results being fetched), as they require ranked results, so I think they should be informative. What do you think?

Reply:
I will check the JDBC implementation and will check if the tacing result is informative for the two methods.

Reply:
Your assumption was correct. The search happens in the `scroll` or `iterate` method and only the results are lazy loaded when requested.